nixos/tests/cgit.nix

```diff
@@ -0,0 +1,58 @@
+import ./make-test-python.nix ({ pkgs, ...} : {
+  name = "cgit";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ schnusch ];
+  };
+
+  nodes = {
+    server = { ... }: {
+      services.cgit."localhost" = {
+        enable = true;
+        nginx.location = "/(c)git/";
+        repos = {
+          some-repo = {
+            path = "/srv/git/some-repo";
+            desc = "some-repo description";
+          };
+        };
+      };
+
+      environment.systemPackages = [ pkgs.git ];
+    };
+  };
+
+  testScript = { nodes, ... }: ''
+    start_all()
+
+    server.wait_for_unit("nginx.service")
+    server.wait_for_unit("network.target")
+    server.wait_for_open_port(80)
+
+    server.succeed("curl -fsS http://localhost/%28c%29git/robots.txt")
+
+    server.succeed(
+        "curl -fsS http://localhost/%28c%29git/ | grep -F 'some-repo description'"
+    )
+
+    server.fail("curl -fsS http://localhost/robots.txt")
+
+    server.succeed(
+        "git init --bare -b master /srv/git/some-repo"
+        " && git init -b master reference"
+        " && (cd reference"
+        " && git remote add origin /srv/git/some-repo"
+        " && date > date.txt"
+        " && git add date.txt"
+        " && git -c user.name=test -c user.email=test@localhost commit -m 'add date'"
+        " && git push -u origin master)"
```

Comment:
maybe something like this would be cooler?
```
"${pkgs.writeScript "test-git" ''
  #!/bin/sh
  set -efu
  git init --bare -b matser /srv/git/some-repo
  git init -b master reference
  ...
''}"
```